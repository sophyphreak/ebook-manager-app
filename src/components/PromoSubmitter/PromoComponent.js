import React from 'react';
import {
  Card,
  CardBody,
  CardHeader,
  Col,
  Row
} from 'reactstrap';

import BookDetailsOne from "../../components/PromoSubmitter/PromoPages/BookDetailsOne";
import BookDetailsTwo from "../../components/PromoSubmitter/PromoPages/BookDetailsTwo";
import SiteSelection from "../../components/PromoSubmitter/PromoPages/SiteSelection";
import DateSelection from "../../components/PromoSubmitter/PromoPages/DateSelection";

const PromoComponent = ({
  // Variables
  currentPage,
  error,

  // BookDetailsOne
  title,
  asin,
  amazonURL,
  regPrice,
  isFiction,
  genre,
  subGenre,
  firstName,
  lastName,
  email,

  // BookDetailsTwo
  description,
  authorBio,

  // SiteSelection
  promoType,
  promoSites,

  // DateSelection
  startDate,
  endDate,
  calendarFocus,

  // Methods
  onClickBack,

  // BookDetailsOne
  onTitleChange,
  onAsinChange,
  onAmazonURLChange,
  onRegPriceChange,
  onIsFictionChange,
  onGenreChange,
  onSubGenreChange,
  onFirstNameChange,
  onLastNameChange,
  onEmailChange,
  onSubmitBookDetailsOne,

  // BookDetailsTwo
  onDescriptionChange,
  onAuthorBioChange,
  onSubmitBookDetailsTwo,

  // SiteSelection
  onPromoTypeChange,
  onPromoSitesChange,
  onSubmitSiteSelection,

  // DateSelection
  onDatesChange,
  onFocusChange,
  onSubmitDateSelection
}) => (
  <Row>
    <Col sm="3" />
    <Col xs="12" sm="6">
      <Card>
        <CardHeader>
          <strong>Promotion Submitter</strong>
        </CardHeader>
        <CardBody>
          {currentPage === "BookDetailsOne" && (
            <BookDetailsOne
              error={error}
              title={title}
              asin={asin}
              regPrice={regPrice}
              amazonURL={amazonURL}
              isFiction={isFiction}
              genre={genre}
              subGenre={subGenre}
              firstName={firstName}
              lastName={lastName}
              email={email}

              onTitleChange={onTitleChange}
              onAsinChange={onAsinChange}
              onAmazonURLChange={onAmazonURLChange}
              onRegPriceChange={onRegPriceChange}
              onIsFictionChange={onIsFictionChange}
              onGenreChange={onGenreChange}
              onSubGenreChange={onSubGenreChange}
              onFirstNameChange={onFirstNameChange}
              onLastNameChange={onLastNameChange}
              onEmailChange={onEmailChange}
              onSubmit={onSubmitBookDetailsOne}
            />
          )}
          {currentPage === "BookDetailsTwo" && (
            <BookDetailsTwo
              error={error}
              description={description}
              authorBio={authorBio}

              onDescriptionChange={onDescriptionChange}
              onAuthorBioChange={onAuthorBioChange}
              onClickBack={onClickBack}
              onSubmit={onSubmitBookDetailsTwo}
            />
          )}
          {currentPage === "SiteSelection" && (
            <SiteSelection
              error={error}
              promoType={promoType}
              promoSites={promoSites}
              
              onPromoTypeChange={onPromoTypeChange}
              onPromoSitesChange={onPromoSitesChange}
              onClickBack={onClickBack}
              onSubmit={onSubmitSiteSelection}
            />
          )}
          {currentPage === "DateSelection" && (
            <DateSelection
              startDate={startDate}
              endDate={endDate}
              focusedInput={calendarFocus}

              onDatesChange={onDatesChange}
              onFocusChange={onFocusChange}
              onClickBack={onClickBack}
              onSubmit={onSubmitDateSelection}
            />
          )}
          {currentPage === "SubmissionSuccess" && (
            <p className="animated fadeIn">
              Your Promo Submission has been submitted. Please wait for a
              confirmation email.
                </p>
          )}
        </CardBody>
      </Card>
    </Col>
  </Row>
);

export default PromoComponent;