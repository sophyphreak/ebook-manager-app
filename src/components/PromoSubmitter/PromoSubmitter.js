import React, { Component } from 'react';

import AmazonURL from '../FormElements/AmazonURL';
import ASIN from '../FormElements/ASIN';
import AuthorBio from '../FormElements/AuthorBio';
// import Cover from '../FormElements/Cover';
import CurrentPrice from '../FormElements/CurrentPrice';
import Description from '../FormElements/Description';
import Email from '../FormElements/Email';
import EndDate from '../FormElements/EndDate';
import FirstName from '../FormElements/FirstName';
import Genre from '../FormElements/Genre';
import LastName from '../FormElements/LastName';
import NumReviews from '../FormElements/NumReviews';
import PressRelease from '../FormElements/PressRelease';
import RegPrice from '../FormElements/RegPrice';
import ReviewAvg from '../FormElements/ReviewAvg';
import StartDate from '../FormElements/StartDate';
import Title from '../FormElements/Title';
import Website from '../FormElements/Website';

export default class PromoSubmitter extends Component {
  constructor(props) {
    super(props);
    this.state = { 
      ASIN:         '',
      firstName:    '',
      lastName:     '',
      email:        '',
      authorBio:    '',
      startDate:    '',
      endDate:      '',
      title:        '',
      amazonURL:    '',
      description:  '',
      // cover:        '',
      numReviews:   '',
      reviewAvg:    '',
      regPrice:     '',
      currentPrice: '',
      genre:        '',
      website:      '',
      pressRelease: '',
      error: ''
    };

    this.onASINChange = this.onASINChange.bind(this);
    this.onFirstNameChange = this.onFirstNameChange.bind(this);
    this.onLastNameChange = this.onLastNameChange.bind(this);
    this.onEmailChange = this.onEmailChange.bind(this);
    this.onAuthorBioChange = this.onAuthorBioChange.bind(this);
    this.onStartDateChange = this.onStartDateChange.bind(this);
    this.onEndDateChange = this.onEndDateChange.bind(this);
    this.onTitleChange = this.onTitleChange.bind(this);
    this.onAmazonURLChange = this.onAmazonURLChange.bind(this);
    this.onDescriptionChange = this.onDescriptionChange.bind(this);
    // this.onCoverChange = this.onCoverChange.bind(this);
    this.onNumReviewsChange = this.onNumReviewsChange.bind(this);
    this.onReviewAvgChange = this.onReviewAvgChange.bind(this);
    this.onRegPriceChange = this.onRegPriceChange.bind(this);
    this.onCurrentPriceChange = this.onCurrentPriceChange.bind(this);
    this.onGenreChange = this.onGenreChange.bind(this);
    this.onWebsiteChange = this.onWebsiteChange.bind(this);
    this.onPressReleaseChange = this.onPressReleaseChange.bind(this);
    this.onSubmit = this.onSubmit.bind(this);
  }

  onASINChange(e) {
    const ASIN = e.target.value.toUpperCase();
    if (ASIN.match(/^[0-9A-Z]{0,10}$/)) {
      this.setState(() => ({ ASIN }));
    }
  };
  
  onFirstNameChange(e) {
    const firstName = e.target.value;
    this.setState(() => ({ firstName }));
  };

  onLastNameChange(e) {
    const lastName = e.target.value;
    this.setState(() => ({ lastName }));
  };

  onEmailChange(e) {
    const email = e.target.value;
    this.setState(() => ({ email }));
  };

  onAuthorBioChange(e) {
    const authorBio = e.target.value;
    this.setState(() => ({ authorBio }));
  };

  onStartDateChange(e) {
    const startDate = e.target.value;
    this.setState(() => ({ startDate }));
  };

  onEndDateChange(e) {
    const endDate = e.target.value;
    this.setState(() => ({ endDate }));
  };

  onTitleChange(e) {
    const title = e.target.value;
    this.setState(() => ({ title }));
  };

  onAmazonURLChange(e) {
    const amazonURL = e.target.value;
    this.setState(() => ({ amazonURL }));
  };

  onDescriptionChange(e) {
    const description = e.target.value;
    this.setState(() => ({ description }));
  };

  // onCoverChange(e) {
  //   const cover = e.target.value;
  //   this.setState(() => ({ cover }));
  // };

  onNumReviewsChange(e) {
    const numReviews = e.target.value;
    this.setState(() => ({ numReviews }));
  };

  onReviewAvgChange(e) {
    const reviewAvg = e.target.value;
    if (!reviewAvg || reviewAvg.match(/^[0-5](\.\d{0,1})?$/)) {   
      this.setState(() => ({ reviewAvg }));
    }
  };

  onRegPriceChange(e) {
    const regPrice = e.target.value;
    if (!regPrice || regPrice.match(/^\d{1,}(\.\d{0,2})?$/)) {
      this.setState(() => ({ regPrice }));
    }
  };

  onCurrentPriceChange(e) {
    const currentPrice = e.target.value;
    if (!currentPrice || currentPrice.match(/^\d{1,}(\.\d{0,2})?$/)) {
      this.setState(() => ({ currentPrice }));
    }
  };

  onGenreChange(e) {
    const genre = e.target.value;
    this.setState(() => ({ genre }));
  };

  onWebsiteChange(e) {
    const website = e.target.value;
    this.setState(() => ({ website }));
  };

  onPressReleaseChange(e) {
    const pressRelease = e.target.value;
    this.setState(() => ({ pressRelease }));
  };

  onSubmit(e) {
    e.preventDefault();
    if (
      !this.state.ASIN ||
      !this.state.firstName ||
      !this.state.lastName ||
      !this.state.email ||
      !this.state.authorBio ||
      !this.state.startDate ||
      !this.state.endDate ||
      !this.state.title ||
      !this.state.amazonURL ||
      !this.state.description ||
      !this.state.numReviews ||
      !this.state.regPrice ||
      !this.state.currentPrice ||
      !this.state.genre ||
      !this.state.pressRelease
    ) {
      this.setState(() => ({ error: 'Please fill in all required fields.'}));
    } else if (!this.state.ASIN.match(/^[0-9A-Z]{10}$/)) {
      this.setState(() => ({ error: 'Please provide a valid ASIN.' }))
    } else if (!this.state.email.match(/^([a-z0-9_\.-]+)@([\da-z\.-]+)\.([a-z\.]{2,6})$/)) {
      this.setState(() => ({ error: 'Please provide a valid email address.' }));    
    } else if (!this.state.amazonURL.match(/^(https?:\/\/)?([\da-z\.-]+)\.([a-z\.]{2,6})([\/\w \.-]*)*\/?$/)) {
      // currently accepts any valid URL
      this.setState(() => ({ error: 'Please provide a valid Amazon URL.' }));
    } else if (!this.state.reviewAvg || parseInt(this.state.reviewAvg) > 5) { 
      this.setState(() => ({ error: 'Please provide a Review Avg of 5 or less.' }));      
    } else if (!this.state.website || !this.state.website.match(/^(https?:\/\/)?([\da-z\.-]+)\.([a-z\.]{2,6})([\/\w \.-]*)*\/?$/)) {
      this.setState(() => ({ error: 'Please provide a valid website URL.' }));
    } else {
      this.setState(() => ({ error: '' }));      
      console.log(this.state);
      alert('See console for state. (Ctrl + Shift + i, and then click Console tab)');
    }
  }

  render() {
    return (
      <div>
        <h4>Promotion Submitter</h4>
        <form className="form-layout" onSubmit={this.onSubmit}>
          {this.state.error && <p>{this.state.error}</p>}
          <ASIN value={this.state.ASIN} onChange={this.onASINChange} />
          <FirstName value={this.state.firstName} onChange={this.onFirstNameChange} />
          <LastName value={this.state.lastName} onChange={this.onLastNameChange} />
          <Email value={this.state.email} onChange={this.onEmailChange} />
          <AuthorBio value={this.state.authorBio} onChange={this.onAuthorBioChange} />
          <StartDate value={this.state.startDate} onChange={this.onStartDateChange} />
          <EndDate value={this.state.endDate} onChange={this.onEndDateChange} />
          <Title value={this.state.title} onChange={this.onTitleChange} />
          <AmazonURL value={this.state.amazonURL} onChange={this.onAmazonURLChange} />
          <Description value={this.state.description} onChange={this.onDescriptionChange} />
          {/* <Cover value={this.state.cover} onChange={this.onCoverChange} /> */}
          <NumReviews value={this.state.numReviews} onChange={this.onNumReviewsChange} />
          <ReviewAvg value={this.state.reviewAvg} onChange={this.onReviewAvgChange} />
          <RegPrice value={this.state.regPrice} onChange={this.onRegPriceChange} />
          <CurrentPrice value={this.state.currentPrice} onChange={this.onCurrentPriceChange} />
          <Genre value={this.state.genre} onChange={this.onGenreChange} />
          <Website value={this.state.website} onChange={this.onWebsiteChange} />
          <PressRelease value={this.state.pressRelease} onChange={this.onPressReleaseChange} />
          <input type="submit" value="Submit" />
        </form>
      </div>
    );
  }
};
