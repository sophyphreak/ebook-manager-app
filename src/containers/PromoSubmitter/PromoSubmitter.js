import React, { Component } from "react";
import moment from "moment";
import axios from "axios";

import PromoComponent from '../../components/PromoSubmitter/PromoComponent';

export default class PromoSubmitter extends Component {
  constructor(props) {
    super(props);
    this.state = {
      currentPage: "PromoPage1",
      error: "",

      // PromoPage1
      title: "",
      asin: "",
      amazonURL: "",
      isFiction: "",
      genre: "",
      isGenreDisabled: true,
      subGenre: "",
      firstName: "",
      lastName: "",
      email: "",
      
      // PromoPage2
      price: "",
      promoType: "",
      startDate: moment(),
      endDate: moment(),
      calendarFocus: null,
      
      // PromoPage3
      description: "",
      authorBio: ""
    };

    // PromoPage1
    this.onTitleChange = this.onTitleChange.bind(this);
    this.onAsinChange = this.onAsinChange.bind(this);
    this.onAmazonURLChange = this.onAmazonURLChange.bind(this);
    this.onIsFictionChange = this.onIsFictionChange.bind(this);
    this.onGenreChange = this.onGenreChange.bind(this);
    this.onSubGenreChange = this.onSubGenreChange.bind(this);
    this.onFirstNameChange = this.onFirstNameChange.bind(this);
    this.onLastNameChange = this.onLastNameChange.bind(this);
    this.onEmailChange = this.onEmailChange.bind(this);
    this.onSubmitPromoPage1 = this.onSubmitPromoPage1.bind(this);
    
    // PromoPage2
    this.onPriceChange = this.onPriceChange.bind(this);
    this.onPromoTypeChange = this.onPromoTypeChange.bind(this);
    this.onDatesChange = this.onDatesChange.bind(this);
    this.onFocusChange = this.onFocusChange.bind(this);
    this.onSubmitPromoPage2 = this.onSubmitPromoPage2.bind(this);
    
    // PromoPage3
    this.onDescriptionChange = this.onDescriptionChange.bind(this);
    this.onAuthorBioChange = this.onAuthorBioChange.bind(this);
    this.onSubmitPromoPage3 = this.onSubmitPromoPage3.bind(this);

    this.onBack = this.onBack.bind(this);
  }

  // PromoPage1

  onTitleChange(e) {
    const title = e.target.value;
    this.setState(() => ({ title }));
  }

  onAsinChange(e) {
    const asin = e.target.value.toUpperCase();
    if (asin.match(/^[0-9A-Z]{0,10}$/)) {
      this.setState(() => ({ asin }));
    }
  }

  onAmazonURLChange(e) {
    const amazonURL = e.target.value;
    this.setState(() => ({ amazonURL }));
  }

  onIsFictionChange(e) {
    const isFiction = e.target.value;
    if (isFiction === 'Fiction') {
      const isGenreDisabled = false;
      this.setState(() => ({ isGenreDisabled }));
    } else {
      const isGenreDisabled = true;
      this.setState(() => ({ isGenreDisabled }));
    };
    this.setState(() => ({ isFiction }));
  }

  onGenreChange(e) {
    const genre = e.target.value;
    this.setState(() => ({ genre }));
  }

  onSubGenreChange(e) {
    const subGenre = e.target.value;
    this.setState(() => ({ subGenre }));
  }

  onFirstNameChange(e) {
    const firstName = e.target.value;
    this.setState(() => ({ firstName }));
  }

  onLastNameChange(e) {
    const lastName = e.target.value;
    this.setState(() => ({ lastName }));
  }

  onEmailChange(e) {
    const email = e.target.value;
    this.setState(() => ({ email }));
  }

  onSubmitPromoPage1(e) {
    e.preventDefault();
    const {
      title,
      asin,
      amazonURL,
      isFiction,
      firstName,
      lastName,
      email
    } = this.state;
    if (
      !title ||
      !asin ||
      !amazonURL ||
      !isFiction ||
      !firstName ||
      !lastName ||
      !email
    ) {
      this.setState(() => ({ error: "Please fill in all required fields." }));
    } else if (
      isFiction === 'Fiction' && !genre
    ) {
      this.setState(() => ({ error: "Please select a genre." }));      
    } else if (
      !asin.match(/^[0-9A-Z]{10}$/)
    ) {
      this.setState(() => ({ error: "Please provide a valid ASIN." }));
    } else if (
      !amazonURL.match(/^(http|https?:\/\/)?(www\.)?(amazon\.com)/)
    ) {
      this.setState(() => ({ error: "Please provide a valid Amazon url." }));
    } else if (
      !email.match(/^([a-z0-9_\.-]+)@([\da-z\.-]+)\.([a-z\.]{2,6})$/)
    ) {
      this.setState(() => ({ error: "Please provide a valid email address." }));
    } else {
      console.log(this.state);

      const error = "";
      this.setState(() => ({ error }));

      const currentPage = "PromoPage2";
      this.setState(() => ({ currentPage }));
    }
  }

  // PromoPage2

  onPriceChange(e) {
    const price = e.target.value;
    if (!price || price.match(/^\d{1,}(\.\d{0,2})?$/)) {
      this.setState(() => ({ price }));
    }
  }

  onPromoTypeChange(e) {
    const promoType = e.target.value;
    this.setState(() => ({ promoType }));
  }

  onDatesChange({ startDate, endDate }) {
    this.setState(() => ({ startDate, endDate }));
  }

  onFocusChange(calendarFocus) {
    this.setState(() => ({ calendarFocus }));
  }

  onSubmitPromoPage2(e) {
    e.preventDefault();
    const {
      price,
      promoType,
      startDate,
      endDate
    } = this.state;
    if (
      !price || 
      !promoType ||
      !startDate ||
      !endDate
    ) {
      this.setState(() => ({ error: "Please fill in all required fields." }));
    } else {
      console.log(this.state);

      const error = "";
      this.setState(() => ({ error }));

      const currentPage = "PromoPage3";
      this.setState(() => ({ currentPage }));
    }
  }

  // PromoPage3

  onDescriptionChange(e) {
    const description = e.target.value;
    this.setState(() => ({ description }));
  }

  onAuthorBioChange(e) {
    const authorBio = e.target.value;
    this.setState(() => ({ authorBio }));
  }

  onSubmitPromoPage3(e) {
    e.preventDefault();
    const {
      title,
      asin,
      amazonURL,
      isFiction,
      genre,
      subGenre,
      firstName,
      lastName,
      email,
      description,
      authorBio,
      promoType,
      promoSites,
      startDate,
      endDate
    } = this.state;
    if (
      !description || 
      !authorBio
    ) {
      this.setState(() => ({ error: "Please fill in all required fields." }));
    } else {

      console.log(this.state);

      const error = "";
      this.setState(() => ({ error }));

      const currentPage = "SubmissionSuccess";
      this.setState(() => ({ currentPage }));

      // Currently constructing POST call to server in order to send an email
      axios
        .post("/api/promo_submitter", {
          text: `PROMO SUBMITTER REQUEST

          Title: ${title}
          ASIN: ${asin}
          Amazon URL: ${amazonURL}
          Regular Price ($): ${amazonURL}
          Fiction or Nonfiction?: ${isFiction}
          Genre: ${genre}
          Sub-Genre: ${subGenre}
          Author First Name: ${firstName}
          Author Last Name: ${lastName}
          Email: ${email}
          Description: ${description}
          Author Biography: ${authorBio}
          Promo Type: ${promoType}
          Promo Sites: ${promoSites}
          Start Date: ${startDate.format("MMMM Do YYYY")}
          End Date: ${endDate.format("MMMM Do YYYY")}
        `,
          html: `
          <h3>Promo Submitter Request</h3>
          <p>Title: ${title}</p>
          <p>ASIN: ${asin}</p>
          <p>Amazon URL: ${amazonURL}</p>
          <p>Regular Price ($): ${price}</p>
          <p>Fiction or Nonfiction?: ${isFiction}</p>
          <p>Genre: ${genre}</p>
          <p>Sub-Genre: ${subGenre}</p>
          <p>Author First Name: ${firstName}</p>
          <p>Author Last Name: ${lastName}</p>
          <p>Email: ${email}</p>
          <p>Description: ${description}</p>
          <p>Author Biography: ${authorBio}</p>
          <p>Promo Type: ${promoType}</p>
          <p>Promo Sites: ${promoSites}</p>
          <p>Start Date: ${startDate.format("MMMM Do YYYY")}</p>
          <p>End Date: ${endDate.format("MMMM Do YYYY")}</p>
        `
        })
        .then(res => {
          console.log(res);
        })
        .catch(err => {
          console.log(err);
        });
    }
  }

  onBack(e) {
    if (this.state.currentPage === "PromoPage2") {
      const currentPage = "PromoPage1";
      this.setState(() => ({ currentPage }));
    }
    if (this.state.currentPage === "PromoPage3") {
      const currentPage = "PromoPage2";
      this.setState(() => ({ currentPage }));
    }
  }

  render() {
    const {
      currentPage,
      error,
      title,
      asin,
      amazonURL,
      price,
      isFiction,
      genre,
      isGenreDisabled,
      subGenre,
      firstName,
      lastName,
      email,
      description,
      authorBio,
      promoType,
      startDate,
      endDate,
      calendarFocus
    } = this.state;
    return (
      <PromoComponent 
        // State
        currentPage={currentPage}
        error={error}

        // PromoPage1
        title={title}
        asin={asin}
        amazonURL={amazonURL}
        isFiction={isFiction}
        genre={genre}
        isGenreDisabled={isGenreDisabled}
        subGenre={subGenre}
        firstName={firstName}
        lastName={lastName}
        email={email}
        
        // PromoPage2
        price={price}
        promoType={promoType}
        startDate={startDate}
        endDate={endDate}
        calendarFocus={calendarFocus}
        
        // PromoPage3
        description={description}
        authorBio={authorBio}

        // Methods
        onBack={this.onBack}

        // PromoPage1
        onTitleChange={this.onTitleChange}
        onAsinChange={this.onAsinChange}
        onAmazonURLChange={this.onAmazonURLChange}
        onIsFictionChange={this.onIsFictionChange}
        onGenreChange={this.onGenreChange}
        onSubGenreChange={this.onSubGenreChange}
        onFirstNameChange={this.onFirstNameChange}
        onLastNameChange={this.onLastNameChange}
        onEmailChange={this.onEmailChange}
        onSubmitPromoPage1={this.onSubmitPromoPage1}
        
        // PromoPage2
        onPriceChange={this.onPriceChange}
        onPromoTypeChange={this.onPromoTypeChange}
        onDatesChange={this.onDatesChange}
        onFocusChange={this.onFocusChange}
        onSubmitPromoPage2={this.onSubmitPromoPage2}
        
        // PromoPage3
        onDescriptionChange={this.onDescriptionChange}
        onAuthorBioChange={this.onAuthorBioChange}
        onSubmitPromoPage3={this.onSubmitPromoPage3}
        />
    );
  }
}
