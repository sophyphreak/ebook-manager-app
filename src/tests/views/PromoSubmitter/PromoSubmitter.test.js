import React from 'react';
import { shallow } from 'enzyme';
import PromoSubmitter from '../../../views/PromoSubmitter/PromoSubmitter';

let wrapper;

beforeEach(() => {
  Date.now = jest.fn(() => 1487076708000)
  wrapper = shallow(<PromoSubmitter />);
});

test('should render PromoSubmitter correctly', () => {
  expect(wrapper).toMatchSnapshot();
});