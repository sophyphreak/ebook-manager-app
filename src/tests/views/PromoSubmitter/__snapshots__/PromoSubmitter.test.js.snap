// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render PromoSubmitter correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PromoSubmitter />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h4>
          Promotion Submitter
        </h4>,
        <BookDetailsOne
          ASIN=""
          amazonURL=""
          email=""
          error=""
          firstName=""
          genre=""
          isFiction=""
          lastName=""
          onASINChange={[Function]}
          onAmazonURLChange={[Function]}
          onEmailChange={[Function]}
          onFirstNameChange={[Function]}
          onGenreChange={[Function]}
          onIsFictionChange={[Function]}
          onLastNameChange={[Function]}
          onPriceChange={[Function]}
          onSubGenreChange={[Function]}
          onSubmit={[Function]}
          onTitleChange={[Function]}
          price=""
          subGenre=""
          title=""
        />,
        false,
        false,
        false,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Promotion Submitter",
        },
        "ref": null,
        "rendered": "Promotion Submitter",
        "type": "h4",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "ASIN": "",
          "amazonURL": "",
          "email": "",
          "error": "",
          "firstName": "",
          "genre": "",
          "isFiction": "",
          "lastName": "",
          "onASINChange": [Function],
          "onAmazonURLChange": [Function],
          "onEmailChange": [Function],
          "onFirstNameChange": [Function],
          "onGenreChange": [Function],
          "onIsFictionChange": [Function],
          "onLastNameChange": [Function],
          "onPriceChange": [Function],
          "onSubGenreChange": [Function],
          "onSubmit": [Function],
          "onTitleChange": [Function],
          "price": "",
          "subGenre": "",
          "title": "",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      false,
      false,
      false,
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h4>
            Promotion Submitter
          </h4>,
          <BookDetailsOne
            ASIN=""
            amazonURL=""
            email=""
            error=""
            firstName=""
            genre=""
            isFiction=""
            lastName=""
            onASINChange={[Function]}
            onAmazonURLChange={[Function]}
            onEmailChange={[Function]}
            onFirstNameChange={[Function]}
            onGenreChange={[Function]}
            onIsFictionChange={[Function]}
            onLastNameChange={[Function]}
            onPriceChange={[Function]}
            onSubGenreChange={[Function]}
            onSubmit={[Function]}
            onTitleChange={[Function]}
            price=""
            subGenre=""
            title=""
          />,
          false,
          false,
          false,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Promotion Submitter",
          },
          "ref": null,
          "rendered": "Promotion Submitter",
          "type": "h4",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "ASIN": "",
            "amazonURL": "",
            "email": "",
            "error": "",
            "firstName": "",
            "genre": "",
            "isFiction": "",
            "lastName": "",
            "onASINChange": [Function],
            "onAmazonURLChange": [Function],
            "onEmailChange": [Function],
            "onFirstNameChange": [Function],
            "onGenreChange": [Function],
            "onIsFictionChange": [Function],
            "onLastNameChange": [Function],
            "onPriceChange": [Function],
            "onSubGenreChange": [Function],
            "onSubmit": [Function],
            "onTitleChange": [Function],
            "price": "",
            "subGenre": "",
            "title": "",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        false,
        false,
        false,
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
